contracts/Tokens/VAI/VAIController.sol

```diff
@@ -0,0 +1,759 @@
+     * @param newTreasuryAddress new Treasury address
+     * @param newTreasuryPercent new Treasury percentage
+     * @custom:error ZeroAddressNotAllowed is thrown if newTreasuryGuardian is zero address
+     * @custom:error ZeroAddressNotAllowed is thrown if newTreasuryAddress is zero address
+     * @custom:error CallerNotAuthorized is thrown if caller is not owner or treasury guardian
+     * @custom:error TreasuryPercentOverflow is thrown if newTreasuryPercent exceeds 1e18
+     * @custom:event NewTreasuryGuardian emits on success
+     * @custom:event NewTreasuryAddress emits on success
+     * @custom:event NewTreasuryPercent emits on success
+     * @custom:access Controlled by Governance and TreasuryGuardian
+     */
+    function setTreasuryData(
+        address newTreasuryGuardian,
+        address newTreasuryAddress,
+        uint256 newTreasuryPercent
+    ) external {
+        // Check caller is owner
+        if (!(msg.sender == owner() || msg.sender == treasuryGuardian)) {
+            revert CallerNotAuthorized();
+        }
+
+        ensureNonzeroAddress(newTreasuryGuardian);
+        ensureNonzeroAddress(newTreasuryAddress);
+
+        if (newTreasuryPercent > 1e18) {
+            revert TreasuryPercentOverflow();
+        }
+
+        emit NewTreasuryGuardian(treasuryGuardian, newTreasuryGuardian);
+        emit NewTreasuryAddress(treasuryAddress, newTreasuryAddress);
+        emit NewTreasuryPercent(treasuryPercent, newTreasuryPercent);
+
+        treasuryGuardian = newTreasuryGuardian;
+        treasuryAddress = newTreasuryAddress;
+        treasuryPercent = newTreasuryPercent;
+    }
+
+    /**
+     * @notice Accrues Vai Interest
+     */
+    function accrueVAIInterest() public {
+        uint256 delta = getVAIRepayRatePerBlock() * (_getBlockNumber() - accrualBlockNumber);
+        delta += VAIMintIndex;
+
+        VAIMintIndex = delta;
+        accrualBlockNumber = _getBlockNumber();
+    }
+
+    /**
+     * @notice This function calculates the VAI amount that can be minted to the minter
+     * @param minter Address of the minter
+     * @custom:error ZeroValueNotAllowed is thrown if oraclePrice is 0
+     * @return (uint256) Returns the mintable VAI amount
+     */
+    // solhint-disable-next-line code-complexity
+    function getMintableVAI(address minter) public view returns (uint256) {
+        if (mintEnabledOnlyForPrimeHolder && !IPrime(prime).isUserPrimeHolder(minter)) {
+            return 0;
+        }
+
+        ResilientOracleInterface oracle = comptroller.oracle();
+        VToken[] memory enteredMarkets = comptroller.getAssetsIn(minter);
+
+        AccountAmountLocalVars memory vars; // Holds all our calculation results
+
+        uint256 accountMintableVAI;
+        uint256 numberOfMarkets = enteredMarkets.length;
+
+        /**
+         * We use this formula to calculate mintable VAI amount.
+         * totalSupplyAmount * VAIMintRate - (totalBorrowAmount + mintedVAIOf)
```

Comment:
I wonder if we use the formula without the mint rate, i.e.

```suggestion
         * weightedSupplyAmount - (totalBorrowAmount + mintedVAIOf)
```

Reply:
I am not able to understand it properly.
How can we calculate the `weightedSupplyAmount`?

Reply:
Oh, it's just a different word for totalSupplyAmount in this formula, sorry for the confusion. I used a different term because in IL Comptroller total supply = sum(amount * price), while weighted supply is either sum(amount * liquidation threshold * price) or sum(amount * collateral factor * price). Here it's the latter, so I figured "weighted supply" term suits better.

Coming back to VAIMintRate, I still can't imagine a use case for setting it below 100%, so the question was whether we could just assume it's always equal to 100%.